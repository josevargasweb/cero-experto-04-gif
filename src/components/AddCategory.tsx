import { useState } from "react";

interface AddCategoryProps {
  // setCategories: React.Dispatch<React.SetStateAction<string[]>>
  onNewCategory: (value: string) => void;
}

/**
 * Component that renders a form to add a new category
 * @param {React.Dispatch<React.SetStateAction<string[]>>} setCategories
 *    Function to update the categories state
 * @returns {JSX.Element} The component
 */
export const AddCategory: React.FC<AddCategoryProps> =  ({onNewCategory}) => {

  const [inputValue, setInputValue] = useState<string>('One Punch');


  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setInputValue(event.target.value);
  }

  const onSubmit = (ev : React.FormEvent) => {
    ev.preventDefault();
    if(inputValue.trim().length <= 1) return;
    onNewCategory(inputValue.trim());
    setInputValue('');
  }

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />

    </form>
  )
}
