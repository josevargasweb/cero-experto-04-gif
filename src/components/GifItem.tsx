import { GiphyImage } from '../models/giphy.models';

export const GifItem: React.FC<GiphyImage> = ({ image, title }) => {
  return (
    <div className="card">
      <img src={image} alt={title} />
      <p>{title}</p>
    </div>
  )
}