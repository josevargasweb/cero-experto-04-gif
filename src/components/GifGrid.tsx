import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

interface GifGridProps {
  category: string
}

export const GifGrid: React.FC<GifGridProps> = ({ category }) => {

  const { images, isLoading } = useFetchGifs(category);



  return (
    <>
      <h3>{category}</h3>
      {
        isLoading
          ? (<h2>Cargando...</h2>)
          :
          <div className="card-grid">
            {
              images.map((img) => (
                <GifItem
                  key={img.id}
                  {...img}
                />
              ))
            }
          </div>
      }


    </>
  )
}
