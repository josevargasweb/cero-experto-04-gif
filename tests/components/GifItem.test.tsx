import React from 'react';
import { describe, expect, test } from 'vitest';
import { render,screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => { 
  const title = "Goku";
  const url = "https://goku.com/goku.jpg";

  test('debe de hacer match con el snapshot', () => { 
    const { container } = render(<GifItem id={'1'} title={title} image={url} />);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
    render(<GifItem id={'1'} title={title} image={url} />);
    
    const { src, alt } = screen.getByRole('img') as HTMLImageElement;

    expect(src).toBe(url);
    expect(alt).toBe(alt);
    
  });

  test('debe de mostrar el titulo en el componente', () => { 

    render(<GifItem id={'1'} title={title} image={url} />);
    expect(screen.getByText(title)).toBeTruthy();
   })
});