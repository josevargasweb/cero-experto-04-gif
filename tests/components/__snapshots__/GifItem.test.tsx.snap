// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Pruebas en <GifItem /> > debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://goku.com/goku.jpg"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
